app/views/MainView/index.tsx

```diff
@@ -29,6 +38,82 @@ const PatientsView = lazy(() => import('../PatientsView'));
 const TemplateView = lazy(() => import('../TemplateView'));
 const ProjectsView = lazy(() => import('../ProjectsView'));
 
+// What fraction of TIME ELAPSED should the user be notified of expiring token
+const TIMEOUT_FRACTION = 0.9;
+
+type TimeoutModalPropTypes = {
+  authorizationToken: string;
+  setAuthorizationToken: React.Dispatch<React.SetStateAction<string>>;
+};
+
+const TimeoutModal = memo(({ authorizationToken, setAuthorizationToken }: TimeoutModalPropTypes) => {
+  const { location: { key: locationKey } } = useHistory();
+  const [open, setIsOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const timerRef = useRef(null);
+
+  // Refresh token everytime user navigates the page, much cleaner than if they did ANY action
```

Comment:
makes sense, but does it slow things down detectably?

Reply:
Haven't tested it actually for when user does `anything` on the front-end

But as you can see, the async calls needs to happen, and if we do cleanup before the async calls goes through, there is still the overhead of attaching and detaching `eventlisteners` and cancelling calls more often than not

Of course, we can make the `updateToken` call to have a parameter (in seconds I believe) before the token expires to refresh 

But IMO, what we have currently is my suggestion on how this interaction would make most sense